### PR TITLE
Check installed mongo extension to use.

### PR DESCRIPTION
**Pull request by Orgun109uk**
Updated MongoDb driver to check if using the legacy driver or not.

### CODE REVIEW

File: src/Codeception/Lib/Driver/MongoDb.php [hunk 1]
```diff
@@ -59,12 +126,7 @@ public function __construct($dsn, $user, $password)
             ];
         }
 
-        try {
-            $this->client = new \MongoClient($dsn, $options);
-            $this->dbh    = $this->client->selectDB($this->dbName);
-        } catch (\MongoConnectionException $e) {
-            throw new ModuleException($this, sprintf('Failed to open Mongo connection: %s', $e->getMessage()));
-        }
+        $this->{$this->legacy ? 'setupMongo' : 'setupMongoDB'}($dsn, $options);
```
Comment (DavertMik) [inline src/Codeception/Lib/Driver/MongoDb.php]: what a strange language construct

**Comment by DavertMik** [comment]
Ok, actually I could merge it but could you fix Travis build? Looks like some dependencies are not installed

File: src/Codeception/Lib/Driver/MongoDb.php (hunk 1 above)
Comment (Orgun109uk) [inline src/Codeception/Lib/Driver/MongoDb.php]: Yeah, an old trick i picked up some time ago.

**Comment by Orgun109uk** [comment]
Have made the mongodb library as a suggestion instead of a requirement.

**Comment by DavertMik** [comment]
You can add `mongodb` to require-dev so it was installed on Travis. 
From what I see, MongoDb tests are skipped. Could you make mongodb package installed and all mongo tests passed on Travis?

**Comment by Orgun109uk** [comment]
Yep, can do. This will require installing the mongodb extension via pecl, if your happy for me to do this?

**Comment by Orgun109uk** [comment]
@DavertMik - I have finally managed to get all builds building successfully, with the exception for "semaphoreci" which appears will require the mongodb extension to be enabled within its settings.

File: composer.json [hunk 2]
```diff
@@ -40,7 +40,9 @@
         "pda/pheanstalk": "~3.0",
         "flow/jsonpath": "~0.2",
         "league/factory-muffin": "^3.0",
-        "league/factory-muffin-faker": "^1.0"
+        "league/factory-muffin-faker": "^1.0",
+        "ext-mongodb": "*",
```
Comment (DavertMik) [inline composer.json]: nah, codeception by itself does not require mongodb. I will remove this constraint

**Comment by DavertMik** [comment]
Thanks! I think we can go with it. Really great stuff

**Pull request merged**